packages/git/src/browser/git-navigable-list-widget.tsx

```diff
@@ -75,14 +76,20 @@ export abstract class GitNavigableListWidget<T extends { selected?: boolean }> e
 
     protected relativePath(uri: URI | string): string {
         const parsedUri = typeof uri === 'string' ? new URI(uri) : uri;
-        const repo = this.repositoryProvider.selectedRepository;
+        const repo = this.repositoryProvider.findRepository(parsedUri);
         if (repo) {
             return Repository.relativePath(repo, parsedUri).toString();
         } else {
             return this.labelProvider.getLongName(parsedUri);
         }
     }
 
+    protected getRepositoryLabel(uri: string): string | undefined {
+        const repository = this.repositoryProvider.findRepository(new URI(uri));
+        const isSelectedRepo = this.repositoryProvider.selectedRepository && repository && this.repositoryProvider.selectedRepository.localUri === repository.localUri;
+        return repository && !isSelectedRepo ? this.labelProvider.getLongName(new URI(repository.localUri)) : undefined;
```

Comment:
Q: Why do we return with `undefined` if the `uri` argument belongs to the selected repository?

Reply:
I am OK with both, just wanted to understand the requirements.